Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -114,7 +114,16 @@ ExceptionOr<RefPtr<JSC::ArrayBuffer>> CompressionStreamEncoder::compress(const u
     }
 
     while (shouldCompress) {
-        auto output = Vector<uint8_t>(allocateSize);
+        if (allocateSize < startingAllocationSize)
+            return Exception { TypeError, "Out of Memory."_s };
+
+        auto output = Vector<uint8_t>();
+        if (!output.tryReserveInitialCapacity(allocateSize)) {
+            allocateSize /= 4;
+            continue;
```

Comment:
Why not doing the if (allocateSize < startingAllocationSize) within this if check?